shell/browser/electron_javascript_dialog_manager.cc

```diff
@@ -53,18 +55,26 @@ void ElectronJavaScriptDialogManager::RunJavaScriptDialog(
     return std::move(callback).Run(false, std::u16string());
   }
 
-  if (dialog_type != JavaScriptDialogType::JAVASCRIPT_DIALOG_TYPE_ALERT &&
-      dialog_type != JavaScriptDialogType::JAVASCRIPT_DIALOG_TYPE_CONFIRM) {
-    std::move(callback).Run(false, std::u16string());
-    return;
-  }
-
   auto* web_preferences = WebContentsPreferences::From(web_contents);
 
   if (web_preferences && web_preferences->ShouldDisableDialogs()) {
     return std::move(callback).Run(false, std::u16string());
   }
 
+  // We want to allow a second call (which is no-op) if the user calls the
+  // event callback AND doesn't use preventDefault()
+  auto adapted_callback = base::AdaptCallbackForRepeating(std::move(callback));
```

Comment:
`base::AdaptCallbackForRepeating` has been removed upstream. https://chromium-review.googlesource.com/c/chromium/src/+/2941842

You'll need to find a better way of doing this

Reply:
I replaced AdaptCallbackForRepeating with SplitOnceCallback.